app/javascript/permanent_edit.js

```diff
@@ -1,8 +1,9 @@
+//! This whole file seems to be isolated from other files, suggesting deletion.
 function changeURL () {
       var week = document.getElementById("cycle_week").value;
       var day = document.getElementById("cycle_day").value;
       var url = window.location.href.slice(-14);
-      if (url == "permanent_edit" || url == "ermanent_edit/") {
+      if (url == "permanent_edit" || url == "ermanent_edit/") {//! the second statement seems useless
```

Comment:
they are doing the OR check because they are slicing the last 14 characters off the end of the URI, which may or may not end in a '/' character. If it does, they will get the '/' character instead of the 'p' at the beginning of 'permanent_edit'. Whether this case would only arise if you directly typed the URI I'm not sure, but it seems like a useful check to me.

Reply:
As we are changing the way to do permanent_edit, I think this line will become useless now.

Reply:
```suggestion
      if (url == "permanent_edit" || url == "ermanent_edit/") {//! the second statement seems useless, and as our team (sp20) decided to switch to a different method, this line will become useless
```